pymeasure/instruments/temptronic/temptronic.py

```diff
@@ -0,0 +1,1301 @@
+                'end of all cycles'  : 16,
+                'end of one cycle' : 8, 
+                'end of test'  : 4,
+                'not at temperature' : 2, 
+                'at temperature'  : 1},
+        map_values=True
+        )
+
+    set_point_number = Instrument.control(
+        "SETN?", "SETN %g",
+        """Select a setpoint to be the current setpoint.
+        SETN nn -- where n is 0 – 17 when on the Cycle screen.
+        or
+        SETN n – where n is 0 to 2 when on the Operator screen 
+        (0=hot, 1=ambient, 2=cold).
+        NOTE: Use *RST to reset (force) the System to the Cycle screen.
+              Use RSTO to reset (force) the System to the Operator screen.
+        NOTE: SETN arguments 0-11 correspon
+        """,
+        validator=truncated_range,
+        values=[0,17]
+        )
+
+    standard_event_status_code = Instrument.measurement(
+        "%S?",  # same as "*ESR?"
+        """Read the standard event status register.
+
+        bit 7 – power on – not used
+        bit 6 – user request -- not used
+        bit 5 – command error (cme)
+        bit 4 – execution error (exe)
+        bit 3 – device dependent error (dde)
+        bit 2 – query error (qye)
+        bit 1 – request control -- not used
+        bit 0 – operation complete -- not implemented
+        NOTE: The above bits are latched, and are automatically cleared
+        when the register is read.
+        """,
+        values={'operation complete' : 1,   # not implemented
+                'request control' : 2,      # not used
+                'query error (qye)' : 4,
+                'device dependent error (dde)' : 8,
+                'execution error (exe)' : 16,
+                'command error (cme)' : 32,
+                'user request' : 64,        # not used
+                'power on' : 128},          # not used
+        map_values=True
+        )
+
+    standard_event_status_mask = Instrument.control(
+        "ESE?", "ESE %g",
+        """Read the standard event status register.
+
+        bit 7 – power on – not used
+        bit 6 – user request -- not used
+        bit 5 – command error (cme)
+        bit 4 – execution error (exe)
+        bit 3 – device dependent error (dde)
+        bit 2 – query error (qye)
+        bit 1 – request control -- not used
+        bit 0 – operation complete -- not implemented
+        NOTE: The above bits are latched, and are automatically cleared
+        when the register is read.
+        """,
+        validator=truncated_range,
+        values={'operation complete' : 1,   # not implemented
+                'request control' : 2,      # not used
+                'query error (qye)' : 4,
+                'device dependent error (dde)' : 8,
+                'execution error (exe)' : 16,
+                'command error (cme)' : 32,
+                'user request' : 64,        # not used
+                'power on' : 128},          # not used
+        map_values=True
```

Comment:
I'm wondering, is the value that is returned by the instrument actually one of the values in the dict (i.e 2, 4, 8, 16 etc) or is it a value between 0 and 255?

Reply:
updated file